### ISSUE TITLE
What is the purpose of bernoulli distribution?

### ISSUE THREAD

**Issue by kkellyk** [LeeHyuck/ABCtorch#5]
Hi, thanks for this great work!
I want to ask about bernoulli distribution in code.
What is the purpose of bernoulli distribution?
What is the difference between ir2 and ir22 in the code?
Do maskbalance and maskbalanceu mean to give target weight?

`maskforbalance = torch.bernoulli(torch.sum(targets_x2 * torch.tensor(ir2).cuda(0), dim=1).detach())`

`ir22 = 1 - (epoch / 500) * (1 - ir2)`
`maskforbalanceu = torch.bernoulli(torch.sum(label_u.cuda(0) * torch.tensor(ir22).cuda(0), dim=1).detach())
`

Thanks
Kelly

**Comment by LeeHyuck** [comment]
Section 3.3
To train the ABC to be balanced, we first generate 0/1 mask M (xb) for each labeled data point xb
using a Bernoulli distribution B (·) with the parameter set to be inversely proportional to the number
of data points of each class. 

Section 3.4
To take full advantage of few unlabeled data points with prediction
confidence values that are higher than the confidence threshold τ in the early stage of training, we
gradually decrease the parameter of the Bernoulli distribution B (·) for ub from 1 to NL/Nqb^
, where
qb^ is the one-hot pseudo-label obtained from qb.

I hope you to read the paper in more detail
